tests/phpunit/tests/test-wpml.php

```diff
@@ -467,4 +467,84 @@ public function test_wpml_switch_language() {
 		do_action( 'wpml_switch_language' );
 		$this->assertEquals( 'en', PLL()->curlang->slug );
 	}
+
+	public function test_wpml_get_element_translations_for_posts() {
+		$frontend = new PLL_Frontend( $this->links_model );
+		$GLOBALS['polylang'] = $frontend;
+
+		$en = self::factory()->post->create( array( 'post_title' => 'My post' ) );
+		self::$model->post->set_language( $en, 'en' );
+
+		$fr = self::factory()->post->create( array( 'post_title' => 'Mon article' ) );
+		self::$model->post->set_language( $fr, 'fr' );
+
+		self::$model->post->save_translations( $en, compact( 'fr' ) );
+
+		$expected = array(
+			'en' => (object) array(
+				'translation_id'       => '0',
+				'language_code'        => 'en',
+				'element_id'           => "$en",
+				'source_language_code' => null,
+				'element_type'         => 'post_post',
+				'original'             => '1',
+				'post_title'           => 'My post',
+				'post_status'          => 'publish',
+			),
+			'fr' => (object) array(
+				'translation_id'       => '0',
+				'language_code'        => 'fr',
+				'element_id'           => "$fr",
+				'source_language_code' => 'en',
+				'element_type'         => 'post_post',
+				'original'             => '0',
+				'post_title'           => 'Mon article',
+				'post_status'          => 'publish',
+			),
+		);
+
+		$trid = apply_filters( 'wpml_element_trid', null, $en );
+		$this->assertEqualSets( $expected, apply_filters( 'wpml_get_element_translations', null, $trid ) );
```

Comment:
Since you took care of using numeric strings instead of integers, let's be strict:
```suggestion
		$this->assertSameSets( $expected, apply_filters( 'wpml_get_element_translations', null, $trid ) );
```

Reply:
We can't do that because we are comparing an array of objects. They have the same properties but are not the same objects.

Reply:
Ah! Crap.